File: Tutorial/Onboarding/onboarding.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

/**
 * CodinGame planet is being attacked by slimy insectoid aliens.
 * <---
 * Hint:To protect the planet, you can implement the pseudo-code provided in the statement, below the player.
 **/
int main()
{

    while (1) {
        string enemy1; 
        cin >> enemy1; cin.ignore();
        int dist1; 
        cin >> dist1; cin.ignore();
        string enemy2; 
        cin >> enemy2; cin.ignore();
        int dist2; 
        cin >> dist2; cin.ignore();

        if(dist1 <= dist2) {
             cout << enemy1 << endl;
        }
        else {
            cout << enemy2 << endl;
        }
    }
}

File: Tutorial/Onboarding/onboarding.java
import java.util.*;
import java.io.*;
import java.math.*;

/**
 * CodinGame planet is being attacked by slimy insectoid aliens.
 * <---
 * Hint:To protect the planet, you can implement the pseudo-code provided in the statement, below the player.
 **/
class Player {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        while (true) {
            String enemy1 = in.next();
            int dist1 = in.nextInt(); 
            String enemy2 = in.next(); 
            int dist2 = in.nextInt(); 
			
            if(dist1 <= dist2) {
                System.out.println( enemy1 );
            }
            else {
                System.out.println( enemy2 );
            }
        }
    }
}